core/src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/CloudbreakResourceNameService.java

```
@@ -44,14 +44,14 @@ protected String adjustPartLength(String part) {
             LOGGER.debug("Shortening part name: {}", part);
             shortPart = String.copyValueOf(part.toCharArray(), 0, MAX_PART_LENGTH);
         } else {
-            LOGGER.debug("Part name length OK, no neet to shorten: {}", part);
+            LOGGER.debug("Part name length OK, no need to shorten: {}", part);
         }
         return shortPart;
     }
 
     protected String normalize(String part) {
         LOGGER.debug("Normalizing resource name part: {}", part);
-        String normalized = part;
+        String normalized = part == null ? "null" : part;
```

We shall throw an exception, because -null- in cluster name is ugly